ibis/expr/api.py

```diff
@@ -507,7 +503,15 @@ def asc(expr: ir.Column | str) -> ir.Value:
     ir.ValueExpr
         An expression
     """
-    return _sort_order(expr, "asc")
+    return _deferred_method_call(expr, "asc")
+
+
+def preceding(value) -> ir.Value:
+    return ops.WindowBoundary(value, preceding=True).to_expr()
+
+
+def following(value) -> ir.Value:
```

Comment:
I thought the plan was to make these private to avoid confusion until we know what that boundary operations do when someone calls `.execute()` on them (if anything).

Reply:
This is a way to explicitly mark expressions as preceding or following if the inference in the `window_boundary` rule wouldn't apply, imagine the following scenario:

```python
start = -myexpr  # ops.Netaget(...something)

ibis.window(rows=(start, t.int_col + 10))
# ROWS BETWEEN <myexpr> PRECEDING AND t.int_col + 10 FOLLOWING`
```

we infer the negate operation as a preceding marker, though the user could have thought about `ROWS BETWEEN <-myexpr> FOLLOWING AND t.int_col + 10 FOLLOWING` in which case wrapping the start expr with `ibis.following()` would ensure that:

```python
ibis.window(rows=(ibis.following(start), t.int_col + 10))
# ROWS BETWEEN -myexpr FOLLOWING AND t.int_col + 10 FOLLOWING`
```

Reply:
So `ibis.preceding()` and `ibis.following()` are for forcing the start/end boundaries instead of letting the window boundary rule to figure it out.